components/kubeflow/katib-launcher/component.yaml

```diff
@@ -1,38 +1,22 @@
-name: Kubeflow - Launch Experiment
-description: Kubeflow Experiment launcher
+name: Katib - Launch Experiment
+description: Katib Experiment launcher
 inputs:
-- {name: Experiment Name,           type: String,                       description: 'Experiment name.'}
-- {name: Experiment Namespace,      type: String,   default: kubeflow,  description: 'Experiment namespace.'}
-- {name: Experiment Version,        type: String,   default: v1alpha3,  description: 'Experiment version.'}
-- {name: Max Trial Count,           type: Integer,                      description: 'How many trials can be executed at most.'}
-- {name: Max Failed Trial Count,    type: Integer,  default: 3,         description: 'How many trials can fail at most.'}
-- {name: Parallel Trial Count,      type: Integer,  default: 3,         description: 'How many trials can be running in parallel at most.'}
-- {name: Objective,                 type: JSON,                         description: 'Experiment objective.'}
-- {name: Algorithm,                 type: JSON,                         description: 'Experiment algorithm.'}
-- {name: Trial Template,            type: JSON,                         description: 'Experiment trialTemplate.'}
-- {name: Parameters,                type: JSON,                         description: 'Experiment Parameter configuration.'}
-- {name: Metrics Collector,         type: JSON,   default: '{}',        description: 'Experiment metricsCollector.'}
-- {name: Experiment Timeout Minutes, type: Integer,  default: 1440,     description: 'Time in minutes to wait for the Experiment to complete.'}
-- {name: Delete Finished Experiment, type: Bool,     default: 'True',   description: 'Whether to delete the experiment after it is finished.'}
+- {name: Experiment Name,            type: String,       default: '',        description: 'Experiment name'}
+- {name: Experiment Namespace,       type: String,       default: anonymous, description: 'Experiment namespace'}
+- {name: Experiment Spec,            type: Base64Pickle, default: '',        description: 'Experiment specification to convert in base64 pickle format'}
```

Comment:
Can you use `JsonObject` type instead of `Base64Pickle`? Pickle is unfriendly to non-python components.

Reply:
@Ark-kun Is it possible to convert Object to `JsonObject` without losing nested class objects ?
When I use `JsonObject` and convert `V1beta1ExperimentSpec` to the `dict`, I can't restore the object in the Launcher.
I have to manually convert `dict` to the `V1beta1ExperimentSpec` object structure.

Reply:
AFAIK, Kubernetes is very YAML-oriented and should support JSON. Perhaps there is some problem with conversion to a dictionary. Can you please check the naming style. Maybe the JSON uses snake_case instead of PascalCase.

Reply:
Yes, that was one of the problem with converting objects.
Currently, I am using Katib SDK with [`sanitize_for_serialization`](https://github.com/kubeflow/pipelines/pull/4798/files#diff-902ee8060c048698267b6a04667ea766ff5e527dd2918444c0700d45539c939fR221) to serialise `ExperimentSpec` object to JSON and [`deserialize`](https://github.com/kubeflow/pipelines/pull/4798/files#diff-aa851c30368bc6a6010bbf6fa8289bfa27d1106fabdfa2c726330d3bdeaed79cR99) to de-serialise JSON to `ExperimentSpec` object and submit Experiment.